packages/typespec-go/test/cadlranch/encode/datetimegroup/custom_client_test.go

```diff
@@ -0,0 +1,219 @@
+	v := time.Date(2022, time.August, 26, 14, 38, 0, 0, gmt)
+	resp, err := client.RFC7231(context.Background(), datetimegroup.RFC7231DatetimeProperty{
+		Value: to.Ptr(v),
+	}, nil)
+	require.NoError(t, err)
+	require.NotNil(t, resp.Value)
+	require.WithinDuration(t, v, *resp.Value, 0)
+}
+
+func TestPropertyClientUnixTimestamp(t *testing.T) {
+	client, err := datetimegroup.NewPropertyClient(nil)
+	require.NoError(t, err)
+	v := time.Unix(1686566864, 0)
+	resp, err := client.UnixTimestamp(context.Background(), datetimegroup.UnixTimestampDatetimeProperty{
+		Value: to.Ptr(v),
+	}, nil)
+	require.NoError(t, err)
+	require.NotNil(t, resp.Value)
+	require.WithinDuration(t, v, *resp.Value, 0)
+}
+
+func TestPropertyClientUnixTimestampArray(t *testing.T) {
+	client, err := datetimegroup.NewPropertyClient(nil)
+	require.NoError(t, err)
+	v := []time.Time{
+		time.Unix(1686566864, 0),
+		time.Unix(1686734256, 0),
+	}
+	resp, err := client.UnixTimestampArray(context.Background(), datetimegroup.UnixTimestampArrayDatetimeProperty{
+		Value: v,
+	}, nil)
+	require.NoError(t, err)
+	require.NotNil(t, resp.Value)
+	for i := 0; i < len(v); i++ {
+		require.WithinDuration(t, v[i], resp.Value[i], 0)
+	}
+}
+
+func TestQueryClientDefault(t *testing.T) {
+	client, err := datetimegroup.NewQueryClient(nil)
+	require.NoError(t, err)
+	gmt, err := time.LoadLocation("GMT")
+	require.NoError(t, err)
+	resp, err := client.Default(context.Background(), time.Date(2022, time.August, 26, 18, 38, 0, 0, gmt), nil)
+	require.NoError(t, err)
+	require.Zero(t, resp)
+}
+
+func TestQueryClientRFC3339(t *testing.T) {
+	client, err := datetimegroup.NewQueryClient(nil)
+	require.NoError(t, err)
+	resp, err := client.RFC3339(context.Background(), time.Date(2022, time.August, 26, 18, 38, 0, 0, time.UTC), nil)
+	require.NoError(t, err)
+	require.Zero(t, resp)
+}
+
+func TestQueryClientRFC7231(t *testing.T) {
+	client, err := datetimegroup.NewQueryClient(nil)
+	require.NoError(t, err)
+	gmt, err := time.LoadLocation("GMT")
+	require.NoError(t, err)
+	resp, err := client.RFC7231(context.Background(), time.Date(2022, time.August, 26, 14, 38, 0, 0, gmt), nil)
+	require.NoError(t, err)
+	require.Zero(t, resp)
+}
+
+func TestQueryClientUnixTimestamp(t *testing.T) {
+	client, err := datetimegroup.NewQueryClient(nil)
+	require.NoError(t, err)
+	resp, err := client.UnixTimestamp(context.Background(), time.Unix(1686566864, 0), nil)
+	require.NoError(t, err)
+	require.Zero(t, resp)
+}
+
+func TestQueryClientUnixTimestampArray(t *testing.T) {
+	client, err := datetimegroup.NewQueryClient(nil)
+	require.NoError(t, err)
+	resp, err := client.UnixTimestampArray(context.Background(), []time.Time{
+		time.Unix(1686566864, 0),
+		time.Unix(1686734256, 0),
+	}, nil)
+	require.NoError(t, err)
+	require.Zero(t, resp)
+}
+
+func TestResponseHeaderClientDefault(t *testing.T) {
+	t.Skip("https://github.com/Azure/typespec-azure/pull/6/files?show-viewed-files=true&file-filters%5B%5D=#r1427306977")
```

Comment:
TCGC is incorrectly setting encoding to `rfc3339` for all time header responses.